### PR TITLE
Add optional third position for specifying layout.

### LINKED ISSUE #3: Request layout argument
Would be excellent to add the new `--new-session-with-layout` argument. I am starting with this script:

```fish
#!/usr/bin/env fish

function p # Get git project directory from .projects. See alias repocache.
    set REPO (zoxide query -l | rg --color=never -FxNf ~/.projects | sed s:"$HOME":~: | fzf --reverse)
    if test -n "$REPO"
        string replace '~' $HOME $REPO
    end
end

set -l pdir (p)
if test -z "$pdir"
    exit 1
end

set -l pname (basename $pdir)

if set -q ZELLIJ_SESSION_NAME
    if not string match -q "$pname" "$ZELLIJ_SESSION_NAME"
        zellij pipe --plugin file:~/bin/zellij-switch.wasm -- "$pname::$pdir"
    end
else
    cd $pdir
    if contains $pname (zellij list-sessions -ns)
        if contains $pname (zellij list-sessions -n | rg EXITED | cut -d' ' -f1 | cut -d':' -f2)
            zellij delete-session $pname
            zellij -n editor -s $pname
        else
            zellij attach $pname
        end
    else
        zellij -n editor -s $pname
    end
end
```

And I'd really love to do: `zellij pipe --plugin file:~/bin/zellij-switch.wasm -- -n editor "$pname::$pdir"`

### PR DESCRIPTION
**Pull request by theherk**
Fixes #3 

With this I can use script:

```fish
#!/usr/bin/env fish

function p # Get git project directory from .projects. See alias repocache.
    set REPO (zoxide query -l | rg --color=never -FxNf ~/.projects | sed s:"$HOME":~: | fzf --reverse)
    if test -n "$REPO"
        string replace '~' $HOME $REPO
    end
end

set -l pdir (p)
if test -z "$pdir"
    exit 1
end

set -l pname (basename $pdir)

if set -q ZELLIJ_SESSION_NAME
    if not string match -q "$pname" "$ZELLIJ_SESSION_NAME"
        zellij pipe --plugin file:~/bin/zellij-switch.wasm -- "$pname::$pdir::editor"
    end
else
    cd $pdir
    if contains $pname (zellij list-sessions -ns)
        if contains $pname (zellij list-sessions -n | rg EXITED | cut -d' ' -f1 | cut -d':' -f2)
            zellij delete-session $pname
            zellij -n editor -s $pname
        else
            zellij attach $pname
        end
    else
        zellij -n editor -s $pname
    end
end
```

And launch with a project configuration.

Let me know what you think.

### CODE REVIEW

**Comment by mostafaqanbaryan** [comment]
Hi.

Thank you for this feature!

I'll check it out in a few days.

File: src/main.rs [hunk 1]
```diff
@@ -17,14 +17,26 @@ impl ZellijPlugin for State {
...
+                cwd = Some(PathBuf::from(collection[1]));
+                "default".to_string()
+            }
+            3 => {
+                // New format: session::cwd::layout
+                cwd = Some(PathBuf::from(collection[1]));
+                format!("{}.kdl", collection[2])
+            }
+            _ => "default".to_string(),
+        };
```
Comment (mostafaqanbaryan) [inline src/main.rs]: I think you could handle all of this like before with 2 `if`s:

```
if collection.len() >= 2 {
        cwd = Some(PathBuf::from(collection[1]));
}
if collection.len() == 3 {
        layout_name = format!("{}.kdl", collection[2])
}
```

File: src/main.rs [hunk 2]
```diff
@@ -17,14 +17,26 @@ impl ZellijPlugin for State {
...
+
+        let layout = LayoutInfo::File(layout_name);
+        switch_session_with_layout(Some(session_name), layout, cwd);
```
Comment (mostafaqanbaryan) [inline src/main.rs]: Why the change from `&session_name` to `session_name`?

**Review by mostafaqanbaryan — CHANGES_REQUESTED** [review]

File: src/main.rs (hunk 1 above)
Comment (theherk) [reply src/main.rs]: Yeah, that's a bit simpler and does seem to work as intended.

File: src/main.rs (hunk 2 above)
Comment (theherk) [reply src/main.rs]: With a terrible memory, I can't quite recall why, but I have switched it back.

**Comment by mostafaqanbaryan** [comment]
Squashed and merged in https://github.com/mostafaqanbaryan/zellij-switch/pull/5.

Thank you!
